pkg/multitenant/multitenant.go

```diff
@@ -0,0 +1,24 @@
+package multitenant
+
+import (
+	"context"
+)
+
+type Tenant interface {
+	GetTenantIds(ctx context.Context) ([]string, error)
+	TenantContextKey() any
```

Comment:
The go-idiomatic way to do this is to keep the context key completly internal to the package.

You'd add `WithTenant(ctx context.Context, tenantId string)` and `TenantFromCtx(ctx context.Context) string`

For example: https://gist.github.com/ww9/4ad7b2ddfb94816a30dfdf2218e02f48

Reply:
That way your tenant interface is a single method interface and you can add:

```go
type TenantFn func(ctx context.Context) ([]string, error)
func (t TenantFn) GetTenantIds(ctx context.Context) ([]string, error) {
    return t(ctx)
}
SingleTenant = TenantFn( func(ctx context.Context) ([]string, error) {return []string{""}, nil}
```

Reply:
👍